### PR TITLE
Fix/create edit course

### PR DESCRIPTION
**Pull request by RamiroJGuillen**
# Summary

Now the Course Table refresh correctly after a course is edited or created and the Form displays the corresponding errors.

# Details

* Refactored code to match the create program implementation
* Added error hints to the form
* Fixed the issue where the request fails or the table doesn't update

# Evidence
Screenshot 2024-02-06 at 20 13 39
added space:
Screenshot 2024-02-07 at 00 29 11

### CODE REVIEW

File: src/app/modules/app/pages/courses/components/course-form/course-form.component.ts [hunk 1]
```diff
@@ -50,43 +41,47 @@ export class CourseFormComponent implements OnInit, OnDestroy {
...
+    } else if (control?.hasError('maxlength')) {
+      const errors = control.errors ? control.errors['maxlength'] : null;
+      const maxLength = errors ? errors.requiredLength : 0;
+      return `This field cannot be longer than ${maxLength} characters.`;
+    }
+    return null;
+  }
+
+  public onSubmit(): void {
+    console.log(this.courseForm.valid);
```
Comment (AJN-jalaU) [inline src/app/modules/app/pages/courses/components/course-form/course-form.component.ts]: remove console.log

File: src/app/modules/app/pages/courses/components/course-form/course-form.component.ts [hunk 2]
```diff
@@ -50,43 +41,47 @@ export class CourseFormComponent implements OnInit, OnDestroy {
...
     if (this.courseForm.valid) {
       if (this.data) {
-        this.editCourse(this.courseForm.value);
+        this.editCourse();
       } else {
-        this.createCourse(this.courseForm.value);
+        this.createCourse();
       }
-    }
+    } else this.courseForm.markAllAsTouched();
```
Comment (AJN-jalaU) [inline src/app/modules/app/pages/courses/components/course-form/course-form.component.ts:63]: missing { after else ?

File: src/app/modules/app/pages/courses/courses.component.ts [hunk 3]
```diff
@@ -48,28 +54,37 @@ export class CoursesComponent implements OnInit {
...
-  showDialogFormComponent(course?: Course): void {
-    const dialogRef$ = this.dialogService.show(
-      CourseFormComponent,
-      course ? course : null
-    );
+  public showFormDialog(course?: Course): void {
+    this.dialogService
+      .show(CourseFormComponent, course ?? null)
+      .subscribe(result => {
+        console.log(result);
```
Comment (AJN-jalaU) [inline src/app/modules/app/pages/courses/courses.component.ts]: can we remove the console.log?

File: src/app/modules/app/pages/programs/programs.component.ts [hunk 4]
```diff
@@ -91,6 +91,7 @@ export class ProgramsComponent {
 
   public showCreateDialog(): void {
     this.dialogService.show(ProgramFormComponent).subscribe(result => {
+      console.log(result);
```
Comment (AJN-jalaU) [inline src/app/modules/app/pages/programs/programs.component.ts]: can we remove console.log?

**Review by AJN-jalaU — CHANGES_REQUESTED** [review]

**Review by carlosBarriosCarvajal — APPROVED** [review]

File: src/app/modules/app/pages/courses/components/course-form/course-form.component.ts (hunk 2 above)
Comment (RonJala) [reply src/app/modules/app/pages/courses/components/course-form/course-form.component.ts:63]: No, an if statement (and else afterwards) execute the inmediate next line of code if not followed by curly braces ({})
example:
if(true)
  doSomething();
else
  doSomethingElse()

Comment (AJN-jalaU) [reply src/app/modules/app/pages/courses/components/course-form/course-form.component.ts:63]: you're rigth. I was confused by the } at line 65

**Review by AJN-jalaU — APPROVED** [review]

**Pull request merged**